src/formik/Checkbox.jsx

```diff
@@ -1,20 +1,33 @@
 import React, { forwardRef } from "react";
 
-import { Field } from "formik";
+import { Field, getIn } from "formik";
 import PropTypes from "prop-types";
 
 import CheckboxField from "components/Checkbox";
 
 const Checkbox = forwardRef(({ name, ...rest }, ref) => (
   <Field {...{ name }}>
-    {({ field, meta }) => (
-      <CheckboxField
-        checked={field.value}
-        {...{ ...field, ref }}
-        error={meta.touched ? meta.error : ""}
-        {...rest}
-      />
-    )}
+    {({ field, meta, form }) => {
+      const { status, setStatus } = form;
+      const fieldStatus = getIn(status, name);
+
+      const fieldProps = {
+        ...field,
+        onChange: e => {
+          setStatus({ ...status, [name]: null });
```

Comment:
Let's make sure that the key is removed instead of setting it to `null`. Do the same for all operations.

```suggestion
          setStatus(dissoc(name));
```